bims/api_views/location_site.py

```diff
@@ -289,6 +294,138 @@ def get(self, request):
             return Response(response_data)
 
 
+    def get_biodiversity_data(self, collections):
+        biodiversity_data = {}
+        taxa = self.get_origin_cons_endemsim_data(collections)
+        # If I found more than one class of animal
+        if 'Actinopterygii' not in taxa:
+            print('Hey there, you are not a fish!')
+        else:
+            biodiversity_data['fish'] = {}
+            biodiversity_data['fish']['origin_chart'] = {}
+            biodiversity_data['fish']['cons_status_chart'] = {}
+            biodiversity_data['fish']['endemism_chart'] = {}
+            biodiversity_data['fish']['origin_chart']['data'] = (
+                taxa['Actinopterygii']['origin_chart']['data'])
+            biodiversity_data['fish']['origin_chart']['keys'] = (
+                taxa['Actinopterygii']['origin_chart']['keys'])
+            biodiversity_data['fish']['cons_status_chart']['data'] = (
+                taxa['Actinopterygii']['cons_status_chart']['data'])
+            biodiversity_data['fish']['cons_status_chart']['keys'] = (
+                taxa['Actinopterygii']['cons_status_chart']['keys'])
+            biodiversity_data['fish']['endemism_chart']['data'] = (
+                taxa['Actinopterygii']['endemism_chart']['data'])
+            biodiversity_data['fish']['endemism_chart']['keys'] = (
+                taxa['Actinopterygii']['endemism_chart']['keys'])
+        biodiversity_data['taxa'] = taxa
+
+        biodiversity_data['occurrences'] = [0, 0, 0]
+
+        biodiversity_data['occurrences'][0] = (
+            sum(taxa['Actinopterygii']['occurrences']))
+
+        biodiversity_data['number_of_taxa'] = [0, 0, 0]
+
+
+        biodiversity_data['number_of_taxa'][0] = (
+            len(taxa['Actinopterygii']['number_of_taxa']))
+        biodiversity_data['ecological_condition'] = ['TBA', 'TBA', 'TBA']
+        return biodiversity_data
+
+
+    def get_origin_cons_endemsim_data(self, collections):
+        taxa = {}
+        for model in collections:
+            if not (model.taxonomy.class_name in taxa):
```

Comment:
Why not refactor this into a local var - you use it many many times and it makes your code verbose.

Reply:
```model.taxonomy.class_name```